### PR TITLE
Get other repo attributes updated

### PR DESCRIPTION
**Pull request by alexpits**

### CODE REVIEW

**Comment by eyalbe4** [comment]
I reviewed the code and let my comments inline.

File: README.md [hunk 1]
```diff
@@ -263,6 +263,26 @@ artifactory.repository("RepoName").replications.createOrReplace(replication)
     String result = artifactory.repositories().update(updatedRepository);
 ```
 
+##### Custom Package Type and Properties
+```
+    CustomPackageTypeImpl customPackageType = new CustomPackageTypeImpl("name");
+    CustomRepositorySettingsImpl settings = new CustomRepositorySettingsImpl(customPackageType);
+
```
Comment (alexpits) [inline README.md]: Done

File: api/src/main/java/org/jfrog/artifactory/client/model/Repository.java [hunk 2]
```diff
@@ -27,4 +29,6 @@
     RepositorySettings getRepositorySettings();
 
     XraySettings getXraySettings();
+
```
Comment (alexpits) [inline api/src/main/java/org/jfrog/artifactory/client/model/Repository.java]: Done

File: README.md (hunk 1 above)
Comment (alexpits) [inline README.md]: Done

File: services/src/main/groovy/org/jfrog/artifactory/client/impl/jackson/RepositoryMixIn.java [hunk 3]
```diff
@@ -29,4 +33,6 @@
     @JsonIgnore
     RepositoryType getRclass();
 
+    @JsonAnyGetter
+    Map getCustomProperties();
```
Comment (alexpits) [inline services/src/main/groovy/org/jfrog/artifactory/client/impl/jackson/RepositoryMixIn.java]: DOne

File: services/src/main/groovy/org/jfrog/artifactory/client/model/builder/impl/RepositoryBuilderBase.groovy [hunk 4]
```diff
@@ -20,6 +20,7 @@ abstract class RepositoryBuilderBase<B extends RepositoryBuilder, R extends Repo
     protected String repoLayoutRef
     protected RepositorySettings settings
     protected XraySettings xraySettings
+    protected Map customProperties
```
Comment (alexpits) [inline services/src/main/groovy/org/jfrog/artifactory/client/model/builder/impl/RepositoryBuilderBase.groovy]: Done

File: services/src/main/groovy/org/jfrog/artifactory/client/model/repository/settings/impl/CustomRepositorySettingsImpl.groovy [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package org.jfrog.artifactory.client.model.repository.settings.impl
```
Comment (alexpits) [inline services/src/main/groovy/org/jfrog/artifactory/client/model/repository/settings/impl/CustomRepositorySettingsImpl.groovy]: Done

File: services/src/main/groovy/org/jfrog/artifactory/client/impl/RepositoryHandleImpl.groovy [hunk 6]
```diff
@@ -63,16 +64,33 @@ class RepositoryHandleImpl implements RepositoryHandle {
...
+
+        Set<String> knownKeys = new HashSet<>()
+        knownKeys.addAll(extractProperties(repo))
+        knownKeys.addAll(extractProperties(repo.xraySettings))
+        knownKeys.addAll(extractProperties(repo.repositorySettings))
+
+        customProperties.keySet().removeAll(knownKeys)
+        return customProperties
+    }
+
```
Comment (eyalbe4) [inline services/src/main/groovy/org/jfrog/artifactory/client/impl/RepositoryHandleImpl.groovy]: Let's convert this method from groovy to java.

**Pull request merged**